src/swarm-tui/Swarm/TUI/Model/Event.hs

```diff
@@ -0,0 +1,225 @@
+  | Robot RobotEvent
+  deriving (Eq, Ord, Show)
+
+swarmEvents :: KeyEvents SwarmEvent
+swarmEvents =
+  keyEvents $
+    concat
+      [ embed Main mainEvents
+      , embed REPL replEvents
+      , embed World worldPanelEvents
+      , embed Robot robotPanelEvents
+      ]
+ where
+  embed f = map (fmap f) . keyEventsList
+
+defaultSwarmBindings :: [(SwarmEvent, [Binding])]
+defaultSwarmBindings =
+  concat
+    [ embed Main defaultMainBindings
+    , embed REPL defaultReplBindings
+    , embed World defaultWorldPanelBindings
+    , embed Robot defaultRobotPanelBindings
+    ]
+ where
+  embed = map . first
+
+-- ----------------------------------------------
+--                 MAIN EVENTS
+-- ----------------------------------------------
+
+-- | Main abstract keybinding events while running the game itself.
+--
+-- See 'Swarm.TUI.Controller.MainEventHandler.'.
+data MainEvent
+  = QuitEvent
+  | ViewHelpEvent
+  | ViewRobotsEvent
+  | ViewRecipesEvent
+  | ViewCommandsEvent
+  | ViewMessagesEvent
+  | ViewStructuresEvent
+  | ViewGoalEvent
+  | HideRobotsEvent
+  | ShowCESKDebugEvent
+  | PauseEvent
+  | RunSingleTickEvent
+  | IncreaseTpsEvent
+  | DecreaseTpsEvent
+  | FocusWorldEvent
+  | FocusRobotEvent
+  | FocusREPLEvent
+  | FocusInfoEvent
+  | ToggleCreativeModeEvent
+  | ToggleWorldEditorEvent
+  | ToggleREPLVisibilityEvent
+  deriving (Eq, Ord, Show, Enum, Bounded)
+
+mainEvents :: KeyEvents MainEvent
+mainEvents = allKeyEvents $ \case
+  QuitEvent -> "quit"
+  ViewHelpEvent -> "view help"
+  ViewRobotsEvent -> "view robots"
+  ViewRecipesEvent -> "view recipes"
+  ViewCommandsEvent -> "view commands"
+  ViewMessagesEvent -> "view messages"
+  ViewStructuresEvent -> "view structures"
+  ViewGoalEvent -> "view goal"
+  HideRobotsEvent -> "hide robots"
+  ShowCESKDebugEvent -> "debug CESK"
+  PauseEvent -> "pause"
+  RunSingleTickEvent -> "run single tick"
+  IncreaseTpsEvent -> "increse TPS"
+  DecreaseTpsEvent -> "decrease TPS"
+  FocusWorldEvent -> "focus World"
+  FocusRobotEvent -> "focus Robot"
+  FocusREPLEvent -> "focus REPL"
+  FocusInfoEvent -> "focus Info"
+  ToggleCreativeModeEvent -> "creative mode"
+  ToggleWorldEditorEvent -> "world editor"
+  ToggleREPLVisibilityEvent -> "toggle REPL"
+
+defaultMainBindings :: [(MainEvent, [Binding])]
+defaultMainBindings = allBindings $ \case
+  QuitEvent -> [ctrl 'q']
+  ViewHelpEvent -> [fn 1]
+  ViewRobotsEvent -> [fn 2]
+  ViewRecipesEvent -> [fn 3]
+  ViewCommandsEvent -> [fn 4]
+  ViewMessagesEvent -> [fn 5]
+  ViewStructuresEvent -> [fn 6]
+  ViewGoalEvent -> [ctrl 'g']
+  HideRobotsEvent -> [meta 'h']
+  ShowCESKDebugEvent -> [meta 'd']
+  PauseEvent -> [ctrl 'p']
+  RunSingleTickEvent -> [ctrl 'o']
+  IncreaseTpsEvent -> [ctrl 'x']
+  DecreaseTpsEvent -> [ctrl 'z']
+  FocusWorldEvent -> [meta 'w']
+  FocusRobotEvent -> [meta 'e']
+  FocusREPLEvent -> [meta 'r']
+  FocusInfoEvent -> [meta 't']
+  ToggleCreativeModeEvent -> [ctrl 'v']
+  ToggleWorldEditorEvent -> [ctrl 'e']
```

Comment:
```suggestion
  ToggleWorldEditorEvent -> []
```

I suggest addressing #1962 by having this *not* bound to a key by default.  I think the world editor is still relatively experimental anyway.